### DISCUSSION TITLE
Potential difference between stable and lazer hidden mod fade-in & fade-out

### DISCUSSION THREAD

**Discussion by jazzberry-jam** [ppy/osu#26256] [Q&A]
### Hidden feels different in lazer
After playing lazer for a good while alongside many years of solely playing maps with hidden, I've come to notice hidden somehow feels *different* in lazer, in a difficult-to-pin-down way. Whether it felt too fast or too slow was hard to say, but it did feel more difficult to read, especially when switching between the two clients in the same session. After seeing [this tweet](https://twitter.com/DepressedHuman8/status/1737552185695904102) match how I felt, I wanted to investigate for myself. I know there have been [similar issues](https://github.com/ppy/osu/discussions/24872) like this in the past, so I know it's a possibility.

### Comparison and Methodology
I recorded 2 lossless .mkv clips in both stable and lazer, both at 1080p 300fps, showcasing AR 9, 9.5 and 10 in each client. I made sure to have both clients running the same settings, with the same skin, 0 universal offset, and fullscreen at 1000fps. I converted them to .mp4 using the same encoding preset in Handbrake to edit in Adobe Premiere and match them side-by-side, exporting them at 300fps. Audio from stable is panned to left, and lazer is panned to the right.

I tried to sync/align both clips the absolute best I could with Premiere using the hitsound waveforms in the timeline, however it's important to note I may be fractions of a second off, and I'm unsure how significant that is in this comparison. According to the waveform peaks in [this screenshot](https://cdn.discordapp.com/attachments/709807278999601184/1190570076244549712/image.png?ex=65a247c9&is=658fd2c9&hm=f120da37ea3ff2da876117764b82c7e4e94c54dea1fb1dddf9656c6467a6b125&), with stable audio on the top and lazer on the bottom, it looks like **I'm at most off by a difference of 3 frames, or 10 milliseconds** in this case.

Comparison at 100% speed

https://github.com/ppy/osu/assets/79028708/56c11fc0-68b5-454b-bab8-1ea8432dbc42

[Uncompressed](https://cdn.discordapp.com/attachments/709807278999601184/1190564564278972457/hidden_test_300.mp4?ex=65a242a7&is=658fcda7&hm=a50ac984ac5cbcd462eb62e9bdbc24e331d2335d36df64df64e7f497bd65e1e4&)

Comparison at 38% speed (lower your volume for this, hitsounds are loud)

https://github.com/ppy/osu/assets/79028708/c460d354-270b-4161-89ab-3b8906c8a02a

[Uncompressed](https://cdn.discordapp.com/attachments/709807278999601184/1190564563071017010/hidden_test_300_slow.mp4?ex=65a242a7&is=658fcda7&hm=426e8751dcbceb5ac479091cc551930e5106e637d6428df0728fdd73e1acb580&)

### Observation
Going frame-by-frame, it looks like stable's hidden mod narrowly but consistently fades in and out faster than lazer's. I'm unsure of the difference in any other animation intricacies, if there are any. Additionally, basing the sync between the two clips on their hitsounds may be faulty because of potential differences in latency between the two clients... and while I'm writing this I'm starting to wonder whether or not this is just a display of lazer simply having more latency on my system.

Either way, if anything comes from this, I hope it's further investigation, as it's a thing I've personally feel like I've noticed for a while as a hidden main, especially with 10+ approach rate.

**Comment by peppy** [comment]
Might be to do with a combination of:

- your skin
- object dim being changed in lazer to be more fair across all skins.

Please redo your test using the default skin.
